content/plugin/log/filtering.mdx

```diff
@@ -0,0 +1,73 @@
+---
+page_title: 'Home - Plugin Development: Filtering Log Output'
+description: >-
+  Information on how to filter the logs that your provider outputs, to help
+  find the information you're looking for.
+---
+
+# Filtering Log Output
+
+A lot of different pieces of software are included in a Terraform run: Terraform itself, the SDK or SDKs that providers are built on, and one or more providers. All of these disparate pieces of software can be writing logs, which means that a lot of different logs are being written, and not all of them are always useful in figuring out why something's happening. To help with that, there are various environment variables used to filter log output.
+
+## Valid Levels
+
+Valid levels you can pass to these environment variables are as follows, in order of most verbose to least verbose:
+
+* `TRACE`
+* `DEBUG`
+* `INFO`
+* `WARN`
+* `ERROR`
+* `OFF` (turns off logs for that logger)
+
+As a special case, `JSON` will output logs in a machine-readable format, though the format is not guaranteed to be stable between releases.
+
+## `TF_LOG`
```

Comment:
what if we tried a table here? We could have a heading that says "Logging Environment Variables" and then use a table so people can quickly skim for the ones they want, see what they are applied to, etc. I just think right now it's hard for me to see the big picture and understand what I would want to set and when. 

| Variable | Scope | Description | 
| `TF_LOG` | All Logs | Sets the most verbose log level for Terraform, the SDKs, and all of the providers. It overrides all other logging environment variable settings (e.g., `TF_LOG_PROVIDER`). When reporting bugs to issue trackers, we recommend setting `TF_LOG=TRACE` because (why?). |